plugin/storage/grpc/proto/storage.proto

```diff
@@ -0,0 +1,188 @@
+}
+
+message GetServicesResponse {
+    oneof response {
+        GetServicesSuccess success = 1;
+        GrpcStoragePluginError error = 2;
+    }
+}
+
+message GetOperationsRequest {
+    string service = 1;
+}
+
+message GetOperationsSuccess {
+    repeated string operations = 1;
+}
+
+message GetOperationsResponse {
+    oneof response {
+        GetOperationsSuccess success = 1;
+        GrpcStoragePluginError error = 2;
+    }
+}
+
+message TraceQueryParameters {
+    string service_name = 1;
+    string operation_name = 2;
+    map<string, string> tags = 3;
+    google.protobuf.Timestamp start_time_min = 4 [
+                                                 (gogoproto.stdtime) = true,
+                                                 (gogoproto.nullable) = false
+                                                 ];
+    google.protobuf.Timestamp start_time_max = 5 [
+                                                 (gogoproto.stdtime) = true,
+                                                 (gogoproto.nullable) = false
+                                                 ];
+    google.protobuf.Duration duration_min = 6 [
+                                              (gogoproto.stdduration) = true,
+                                              (gogoproto.nullable) = false
+                                              ];
+    google.protobuf.Duration duration_max = 7 [
+                                              (gogoproto.stdduration) = true,
+                                              (gogoproto.nullable) = false
+                                              ];
+    int32 num_traces = 8;
+}
+
+message FindTracesRequest {
+    TraceQueryParameters query = 1;
+}
+
+message FindTraceIDsRequest {
+    TraceQueryParameters query = 1;
+}
+
+message FindTraceIDsSuccess {
+    repeated bytes trace_id = 1 [
+                                (gogoproto.nullable) = false,
+                                (gogoproto.customtype) = "github.com/jaegertracing/jaeger/model.TraceID",
+                                (gogoproto.customname) = "TraceIDs"
+                                ];
+}
+
+message FindTraceIDsResponse {
+    oneof response {
+        FindTraceIDsSuccess success = 1;
+        GrpcStoragePluginError error = 2;
+    }
+}
+
+message EmptyResponse {
+
+}
+
+message GrpcStoragePluginError {
+    string message = 1;
+}
+
+service GrpcSpanWriterStoragePlugin {
+    // spanstore/Writer
+    rpc WriteSpan(WriteSpanRequest) returns (WriteSpanResponse);
+}
+
+service GrpcSpanReaderStoragePlugin {
+    // spanstore/Reader
+    rpc GetTrace(GetTraceRequest) returns (GetTraceResponse);
+    rpc GetServices(GetServicesRequest) returns (GetServicesResponse);
+    rpc GetOperations(GetOperationsRequest) returns (GetOperationsResponse);
+    rpc FindTraces(FindTracesRequest) returns (stream Trace);
+    rpc FindTraceIDs(FindTraceIDsRequest) returns (FindTraceIDsResponse);
+}
+
+service GrpcDependenciesStoragePlugin {
```

Comment:
any reason we don't want this split into reader and writer, similar to span storage?

Reply:
Happy to do that. Do we consider dependency store writer as out of scope for this changeset?

Reply:
we can skip it for now.